include/nighthawk/common/request_source_plugin.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "envoy/api/api.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+
+#include "nighthawk/common/request_source.h"
+
+namespace Nighthawk {
+
+/**
+ * An interface for different RequestSource plugins.
+ * Uses a plugin-specific config proto.
+ */
+class RequestSourcePlugin : RequestSource {
+public:
+  RequestSourcePlugin(Envoy::Api::Api& api) : api_(api) {}
```

Comment:
Add `const` into `RequestSourcePlugin(const Envoy::Api::Api& api)`?

Reply:
I don't think I can do that, because I need to use the api for "loadFromFile" which doesn't take a const api.